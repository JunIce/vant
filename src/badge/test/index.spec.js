import { mount } from '../../../test';
import Badge from '..';

test('count prop', () => {
  const wrapper = mount(Badge, {
    propsData: {
      count: 10,
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('dot prop', () => {
  const wrapper = mount(Badge, {
    propsData: {
      dot: true,
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('max-count prop', () => {
  const wrapper = mount(Badge, {
    propsData: {
      maxCount: 50,
      count: 51,
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('offset prop', () => {
  const wrapper = mount(Badge, {
    propsData: {
      count: 10,
      offset: [10, 10],
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('show-zero prop', () => {
  const wrapper = mount(Badge, {
    propsData: {
      count: 0,
      showZero: true,
    },
  });
  expect(wrapper).toMatchSnapshot();
});
