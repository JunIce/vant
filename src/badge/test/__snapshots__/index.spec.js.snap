// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`count prop 1`] = `<span class="van-badge"><sup class="van-badge__number van-badge__standalone">10</sup></span>`;

exports[`dot prop 1`] = `<span class="van-badge"><sup class="van-badge__dot van-badge__standalone"></sup></span>`;

exports[`max-count prop 1`] = `<span class="van-badge"><sup class="van-badge__number van-badge__standalone">50+</sup></span>`;

exports[`offset prop 1`] = `<span class="van-badge"><sup class="van-badge__number van-badge__standalone" style="top: 10px; right: 10px;">10</sup></span>`;

exports[`show-zero prop 1`] = `<span class="van-badge"><sup class="van-badge__number van-badge__standalone">0</sup></span>`;
